integration-tests/baas-test-server/baasaas.ts

```diff
@@ -0,0 +1,225 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import assert from "node:assert";
+import gha from "@actions/core";
+
+const BAASAAS_BASE_URL = "https://us-east-1.aws.data.mongodb-api.com/app/baas-container-service-autzb/endpoint/";
+
+function createHeaders(appendKey = false) {
+  const headers: HeadersInit = {
+    "Content-Type": "application/json",
+  };
+  if (appendKey) {
+    const { BAASAAS_KEY } = process.env;
+    assert(BAASAAS_KEY, "Missing BAASAAS_KEY env");
+    headers["apiKey"] = BAASAAS_KEY;
+  }
+  return headers;
+}
+
+type StartedContainer = {
+  id: string;
+};
+
+function assertStartedContainer(value: unknown): asserts value is StartedContainer {
+  assert(typeof value === "object" && value !== null);
+  assert("id" in value && typeof value.id === "string");
+}
+
+type StartContainerOptions =
+  | {
+      githash: string;
+    }
+  | {
+      branch: string;
+    };
+
+export async function startContainer(options: StartContainerOptions) {
+  const url = new URL("startContainer", BAASAAS_BASE_URL);
+  if ("githash" in options) {
+    url.searchParams.append("branch", options.githash);
+  } else if ("branch" in options) {
+    url.searchParams.append("branch", options.branch);
+  }
+  const response = await fetch(url, { method: "POST", headers: createHeaders(true) });
+  assert(response.ok);
+  const json = await response.json();
+  assertStartedContainer(json);
+  return json;
+}
+
+type StopContainerResponse = {
+  /* TODO: Fill in */
+};
+
+function assertStopContainerResponse(value: unknown): asserts value is StopContainerResponse {
+  /* TODO: Fill in */
+}
+
+export async function stopContainer(id: string) {
+  const url = new URL("stopContainer", BAASAAS_BASE_URL);
+  url.searchParams.append("id", id);
+  const response = await fetch(url, { method: "POST", headers: createHeaders(true) });
+  assert(response.ok);
+  const json = await response.json();
+  assertStopContainerResponse(json);
+}
+
+type ContainerStatus = {
+  id: string;
+  lastStatus: "RUNNING" | "PENDING" | string;
+  startedAt: string | Record<never, never>;
```

Comment:
Is there a particular scenario where `startedAt` is an object?

Reply:
Yeah - I believe it's a bug in the service.
@mpobrien have you seen this too? It seems the `startedAt` can be an object, when the container is pending.

Reply:
Yeah, I think this may be possible if the container was created but ECS did not start it yet, which is a gap of usually a few seconds. I can update the code to leave this field undefined when in this state.

Reply:
OK, this should be fixed now as of https://github.com/10gen/baasaas/commit/5171e89423e117d27119a4a5c2e42d0b555c46ca

Reply:
I've changed this to make it an optional string and simply delete the property if it's not a string.

Reply:
@mpobrien I'm not certain that fix actually works? Since the empty object in JS is truthy.

Reply:
That might be an artifact of the serialization, since `this.result.startedAt` is actually a `Date`.  AFAICT the fix works for me, I was able to run `ls` and catch it at the right time with this output:
```[
  {
    "id": "arn:aws:ecs:us-east-1:969505754201:task/baas-container-service2/2db388ed48c54b37aced9cfc3a710b6f",
    "lastStatus": "RUNNING",
    "createdAt": "2024-01-17T21:40:58.474Z",
    "startedAt": "2024-01-17T21:41:00.633Z",
    "creatorId": "656f6a2f5bdfa7ee906ae64a",
    "creatorName": "...",
    "httpUrl": "...",
    "mongoUrl": "...",
    "tags": []
  },
  {
    "id": "arn:aws:ecs:us-east-1:969505754201:task/baas-container-service2/e0a341515dcc4f4bb72124366231eff7",
    "lastStatus": "RUNNING",
    "createdAt": "2024-01-17T21:22:36.399Z",
    "startedAt": "2024-01-17T21:22:37.366Z",
    "creatorId": "65a598d9a9ff26ae5fc14043",
    "creatorName": "realm-js-ci",
    "httpUrl": "...",
    "mongoUrl": "...",
    "tags": []
  },
  {
    "id": "arn:aws:ecs:us-east-1:969505754201:task/baas-container-service2/ee144edcf0a94e338353c8f7f2c5cbbe",
    "lastStatus": "PENDING",
    "createdAt": "2024-01-17T21:42:14.66Z",
    "creatorId": "656f6a2f5bdfa7ee906ae64a",
    "creatorName": "...",
    "tags": []
  }
]